### DISCUSSION TITLE
it shouldn't check for $pkgDir/package.json

### DISCUSSION THREAD

**Discussion by its-dibo** [semantic-release/semantic-release#2454] [Configuration]
## Current behavior

the package `@semantic-release/npm` checks for the existence of `${pkgDir}/package.json` file before performing a release

the folder `pkgDir` may refer to a dist folder that is created in `prePublishOnly` phase and doesn't exist yet.

building the project everytime before running semantic-release is overkilling.

[minimal repreoduction](https://github.com/eng-dibo/tmp-semantic-release)
run `cd package && npm run release` 
don't forget to set `NPM_TOKEN`

## Expected behavior

check for `${pkgDir}/package.json` after running all `pre` hooks, i.e just before running `puplish` and after `prePublishOnly` and `prepack`

## Environment

- **semantic-release** version: 
```
  "@semantic-release/exec": "^6.0.3",
    "@semantic-release/git": "^10.0.1",
    "@semantic-release/npm": "^9.0.1",
    "semantic-release": "^19.0.2",
    "semantic-release-monorepo": "^7.0.5"
```

**Comment by travi** [comment]
> the folder `pkgDir` may refer to a dist folder that is created in `prePublishOnly` phase and doesn't exist yet.

`pkgDir` should not be necessary in the type of scenario that it sounds like you are describing. instead, the built files should be included either through `files` in the `package.json` or by making sure they are not included in the `.npmignore` file. defining `main` or `exports` in the `package.json` helps the consumer use the files without worrying about which directory that the files are located in. i maintain many packages that have a build step like this, triggered by `prepack` (which i recommend over `prePublishOnly`) and have never needed to define `pkgRoot`
